code/forms/LDAPChangePasswordForm.php

```diff
@@ -3,6 +3,55 @@
 class LDAPChangePasswordForm extends ChangePasswordForm {
 
 	/**
+	 * The sole purpose for overriding the constructor is surfacing the username to the user.
+	 */
+	public function __construct($controller, $name, $fields = null, $actions = null) {
+		// We need to create all fields here, because parent::__construct will not create them if we customise.
+		if(!$fields) {
+			$fields = new FieldList();
+
+			// Security/changepassword?h=XXX redirects to Security/changepassword
+			// without GET parameter to avoid potential HTTP referer leakage.
+			// In this case, a user is not logged in, and no 'old password' should be necessary.
+			if(Member::currentUser()) {
+				$fields->push(new PasswordField("OldPassword",_t('Member.YOUROLDPASSWORD', "Your old password")));
+			}
+
+			$fields->push(new PasswordField("NewPassword1", _t('Member.NEWPASSWORD', "New Password")));
+			$fields->push(new PasswordField("NewPassword2", _t('Member.CONFIRMNEWPASSWORD', "Confirm New Password")));
+		}
+		if(!$actions) {
+			$actions = new FieldList(
+				new FormAction("doChangePassword", _t('Member.BUTTONCHANGEPASSWORD', "Change Password"))
+			);
+		}
+
+		parent::__construct($controller, $name, $fields, $actions);
+
+		$member = Member::currentUser();
+		if(!$member) {
+			if(Session::get('AutoLoginHash')) {
+				$member = Member::member_from_autologinhash(Session::get('AutoLoginHash'));
+			}
+
+			// The user is not logged in and no valid auto login hash is available
+			if(!$member) {
+				Session::clear('AutoLoginHash');
+				return $this->controller->redirect($this->controller->Link('login'));
+			}
+		}
+
+		// Get the username.
+		$ldap = Injector::inst()->get('LDAPService')->getUserByGUID($member->GUID, array('samaccountname'));
```

Comment:
If code below here is the sole reason why you override the __constructure, you can call the parent::__constructor and then change $this->Fields ?

The less code we override, the less mucking around we need to do when it comes to security updates